netlogo-core/src/main/nvm/AnonymousProcedure.scala

```diff
@@ -43,7 +45,7 @@ sealed trait AnonymousProcedure {
 
   def bindArgs(b: Binding, args: Array[AnyRef]) {
     var i = 0
-    var n = formals.size
+    val n = formals.length
     while(i < n) {
       b.let(formals(i), args(i))
```

Comment:
If you are still dissatisfied with the performance after these changes, I suspect that `Binding.let` could have an optimized form which took a list of `Let` and did a mass-assignment of bindings. I would expect the performance of `b.let` here to be roughly O(n^2), after optimization it could be O(n). Granted, the reason I wouldn't expect this to make a difference in most cases is because `n` here is very small - most often 0, otherwise often 1 or 2, and seldom more than perhaps 5.